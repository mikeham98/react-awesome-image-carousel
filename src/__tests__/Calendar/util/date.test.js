import { getDate, getTimeFromDate, checkDate } from '../../../app/ImageSlider/util/date';

describe('date', () => {
  describe('getDate', () => {
    it('should return a date without the time', () => {
      const date = new Date(2019, 0, 1, 5, 10, 42, 5);
      expect(getDate(date)).toMatchSnapshot();
    });
    it('should return undefined if date is not valid', () => {
      const date = 'hello world';
      expect(getDate(date)).toBeUndefined();
    });
  });
  describe('getTimeFromDate', () => {
    it('shoulld return 05:10 when a date is passed in', () =>{
      const date = new Date(2019, 0, 1, 5, 10, 42, 5);
      expect(getTimeFromDate(date)).toBe('05:10')
    })
  });
  describe('checkDate', () => {
    it('should return true if the date is valid', () => {
      const date = new Date(2019, 0, 1, 5, 10, 42, 5);
      expect(checkDate(date)).toBeTruthy();
    });
    it('should return false if the date passed in is not valid', () => {
      const date = 'hello world';
      expect(checkDate(date)).toBeFalsy();
    });
  });
});