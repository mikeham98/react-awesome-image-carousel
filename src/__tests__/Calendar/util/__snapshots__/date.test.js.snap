// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date getDate should return a date without the time 1`] = `2019-01-01T00:00:00.000Z`;
